### DISCUSSION TITLE
Deploy on Helium Network?

### DISCUSSION THREAD

**Discussion by ivelin** [ambianic/ambianic-edge#405] [Ideas]
Currently Ambianic Box mainly targets indoor settings with WiFi access. That allows devices to connect to Ambianic UI, send Email notifications and other tasks.

Ambianic Edge uses relatively low bandwidth and power consumption via Raspberry Pi4.
Unlike traditional IP cameras, Ambianic Edge does not stream audio/video media constantly. Instead if only triggers notifications when an event or object of interest is detected.

That prompts the question whether it is feasible to build an outdoor version that uses long range connectivity such as LoRa to enable uses cases such as monitoring remote points of interest without WiFi or Ethernet access to the Internet.

There is an existing proof of concept project that uses on device inference and LoRa to trigger alert when a person is detected:
https://hackaday.io/project/162667-lora-neural-network-security-system

There is also a related discussion on the Helium blockchain network to support IP over LoRa.
https://github.com/helium/HIP/issues/319

**Comment by wolfenhawke** [comment]
LoRA provides the ability to achieve some very long ranges. This combined with the very low power consumption of the hardware being considered here opens the ability for some interesting applications:

1. home roof monitoring (rodents, other interference)
2. environmental monitoring (maybe solar powered)
3. forest camera notification
4. long driveway visitor alarm (e.g. for countryside properties)
5. farm alarms (grazing limits, predators, etc), could be solar powered

As a point of data, my Helium Lora setup has a reach of about 65km (from San Jose to San Francisco).

**Reply by ivelin** [reply]
Very interesting thoughts, @wolfenhawke .

You should chime in the HIP thread:
https://github.com/helium/HIP/issues/319

> LoRA provides the ability to achieve some very long ranges. This combined with the very low power consumption of the hardware being considered here opens the ability for some interesting applications:
> 
> 1. home roof monitoring (rodents, other interference)

This landed on target. I recently had to fix 3 holes on my roof caused by rodents. Would be great if I could setup a system that detects them and maybe flashes bright light to scare them off the room. WiFi doesn't reach that far and I don't feel like drilling more holes for power cables to the roof. Solar powered LoRa connected camera would be awesome here. 

> 2. environmental monitoring (maybe solar powered)
> 3. forest camera notification
> 4. long driveway visitor alarm (e.g. for countryside properties)
> 5. farm alarms (grazing limits, predators, etc), could be solar powered

Very relevant for my dad's farm where he has to worry about several buildings spread out up to a few miles apart. Occasionally foxes sneak in the chicken coop to feast. Sometimes there are people out to steal horses and cattle.

> As a point of data, my Helium Lora setup has a reach of about 65km (from San Jose to San Francisco).

That is crazy good distance. How about bandwidth in both directions? Can you get something like 16 kbps reliably?

**Reply by wolfenhawke** [reply]
Been there with the rodents. This unit works well for Norway rats and squirrels, but having the ability to turn it on via an alarm relay would give it even more applicability. It tends to be bothersome to younger folks who can hear the harmonics.  But varying turn on (only when detects rodents) would make it more functional. https://www.amazon.com/gp/product/B086VSF924/

**Reply by wolfenhawke** [reply]
On the transmit side, Helium is not particularly meant for continuous transmission, but closer to the sensor chirp like what you were proposing. While it can theoretically reach 5.5kbps (not from spec, but from https://lora.readthedocs.io/en/latest/#data-rates, it depends on the spreading factor that Helium uses. Another source, https://www.rfwireless-world.com/Terminology/LoRaWAN-Spreading-factor-Range-and-Data-Rate.html. But yes, I can get reliable connection both directions (I am witnessing that hotspot) at that distance. Note, I am using a tall 9dbi antenna, but it gives you an idea of the capability. I have tested a compact antenna (HNTenna) to have 35km range. A long duck antenna reliably can reach about 20km. I would expect about 10km from a compact duck, but have not tested this.

**Reply by ivelin** [reply]
Very helpful insights, @wolfenhawke. Gives me an idea of the throughput limitations. I will dive deeper in the LoRa networking stack to understand what can be done realistically for apps that rely on IP connectivity. It appears that we are looking for a middle ground where IP is supported, which makes LoRa accessible to the vast majority of mainstream app developers, but the kind of IoT apps they build must be on a lean bps diet and can't be using SaaS design practices verbatim.

**Reply by wolfenhawke** [reply]
This is good direction. Yes. The Helium network would provide a very low cost, ubiquitous data feed of sensors to a data lake. They call it console, but it could also be routed privately to AWS, or Azure, GCS, or a private server if desired. The SAAS design could then be applied to the data lake or location of data ingestion. So, Helium/Lora is trickle feeding sensor hit data. The digital twin would be working on the data at the ingestion site. This would work well. It would also allow much deeper AI workloads to be applied at the cloud. With training pushed back down to the actual edge IOT device for decision filtering.
Two way is allowed so for instance on a hypothetical rodent alarm, the camera could decide on it's own that it sees a target rodent, but then send a very low res image (or target outline) and alarm to the cloud. The cloud application could then send a signal back to turn on the rodent chaser, and then turn it off after a designed period of time.

**Reply by ivelin** [reply]
Nice flows. @dcmartin also has some good ideas for independent living scenarios. Autonomous smart sensors and cameras make intelligent observations such as person in the kitchen at 9am, backyard 3pm, fridge open at 5pm, bathroom at 7pm, etc. Cloud aggregation model receives these data points without any sensitive raw image data and predicts anomalies in daily behavior. I am pushing this a bit further and hoping some of the aggregation could happen at the edge via collaborative resource pooling and federated learning. Thus further removing the privacy concern with cloud data aggregation.

**Reply by ivelin** [reply]
Hunting cameras orver LoRa with Verizon Cell tower connected base station. A 2020 product.
https://www.trailcampro.com/products/copy-of-covert-lora-lb-v3-verizon
